enhancements/insights/prometheus-alerts-for-insights-recommendations.md

```diff
@@ -0,0 +1,268 @@
+  notably Alertmanager).
+
+## Proposal
+
+### User Stories
+
+As a user, I want to be immediately notified about a recommendation that has
+become active on my cluster without having to periodically check the Insights
+tab. I want to receive notifications through my preferred channel (e.g., Slack,
+email) configured via Alertmanager when a recommendation becomes active, and
+another one when it becomes inactive again.
+
+### Why should this be implemented in Insights Operator?
+
+The Insights Operator is already fetching the relevant data from the Smart Proxy,
+and managing Prometheus metrics based on this data. Therefore, it makes sense
+for additional information about recommendations to be implemented alongside
+what already exists.
+
+### API Extensions
+
+### Implementation Details/Notes/Constraints [optional]
+
+Please note that IO refers to the Insights Operator in this context.
+
+1. IO is enabled and periodically uploads archives to the Ingress service,
+   which are then processed by the CCX data pipeline. (standard behavior)
+2. After a successful archive upload, IO waits for the report to become
+   available, fetches it from the Smart Proxy and processes it. (This is an
+   extension of the current behavior, where we only look at the number of
+   active recommendations per severity level.) (needs to be implemented)
+3. In-cluster Prometheus metrics indicating the status of recommendations
+   being active are set accordingly.
+    - If a recommendation was previously registered as active and is present
+      in the latest Insights report, IO performs no additional action for this
+      recommendation (do not repeat alerts for previously spotted recommendations)
+    - If a recommendation was previously registered as active and the latest
+      report no longer contains it, the metric will either be reset or
+      unregistered from the Prometheus instance if possible. (remove alerts for
+      no longer hitting recommendations)
+    - If a recommendation was previously registered as inactive or the metric
+      for this recommendation is absent and the latest Insights report contains
+      it, a metric will be updated/created to indicate that the recommendation
+      is currently hitting. (indicate new recommendations)
+4. The metric will have labels containing the ID, severity (based on the total
+   risk property), a human-readable name, and a link to a detailed description
+   of the recommendation and its remediation steps (Insights Advisor URL
+   generated based on the cluster ID and rule ID by IO) and its value will be
+   0/1 to indicate if the recommendation is inactive/active respectively.
+5. There will be a generic/templated info-level alert present on the cluster,
+   which will present the user with brief information about the hitting
+   recommendation (its human-readable name and severity),
+   and the link for details and remediation steps.
```

Comment:
A preliminary version of the alerting rule would be useful too.

Reply:
Same as with the metric. These two will be established during the PoC development, as we have no experience with exposing more complicated metrics and defining templated alerts, so we do not know yet how much information we will be able to pass.